### DISCUSSION TITLE
I2C scan writes 0 bytes

### DISCUSSION THREAD

**Discussion by andrei-edward-popa** [zephyrproject-rtos/zephyr#42603] [General] [answered]
**Describe the bug**
I'm wondering why the I2C scan example writes 0 bytes of data to test if a slave acknowledge the address. In some cases it happens that we can't scan the bus without sending a byte to see if a slave acknowledge the transfer. This happens for Designware I2C IP for example.

**Comment by henrikbrixandersen** [comment, marked as answer]
The reason for writing 0 bytes and checking for an ACK is that even writing one byte could be a destructive action (think for instance an I2C EEPROM, where writing one byte would change the change the contents of the EEPROM).

**Reply by andrei-edward-popa** [reply]
Yes, you are right, but what about to read 1 byte instead of a write?

**Reply by henrikbrixandersen** [reply]
Some SMBus chips are write-only and will thus not acknowledge a read request.

**Reply by andrei-edward-popa** [reply]
Okay, I understand, thanks for the explanation
